### DISCUSSION TITLE
how to add breadcrumbs in CMS pages & how to get design view for CMS pages or how to get data in static pages

### DISCUSSION THREAD

**Discussion by osamaahmed32** [wintercms/winter#1165] [General]
how can I add breadcrumbs in CMS pages as its coming in static pages and how can I have design view for dynamic pages as I have for static pages with richtexteditor.

or is there any way that I can get data from db in static pages. please suggest

Request for help from winter cms developers community

I have done like this in static pages

```
==
<?php
use XXXX\Refinedproducts\Models\RefinedProducts;

    function onStart()
    {
        $this['refined_products'] = RefinedProducts::where('is_show', '=', '1' )->orderBy('id', 'desc')->get();
    }
?>
==

{{ product.issue_date }}
```

but when go on the page in the browser it shows me {{ product.issue_date }} it doesnt shows me the value of it

Thanks alot in advance

**Comment by LukeTowers** [comment]
My response to your email below:

> 1. You should use the Database Templates feature (https://wintercms.com/docs/v1.2/docs/cms/themes#database-driven-themes). 
> 2. I don't know what you mean about getting a breadcrumb from the CMS section, but it is possible to build your own breadcrumbs with partials (see https://34aircadets.ca/resources/uniform/medals for an example of a Winter CMS page with a breadcrumb). 
> 3. No, don't use the CMS section for non-technical editors. You should use the CMS section to build a page layout and page that the editors can then edit in the Pages section. Look at Custom page fields (https://github.com/wintercms/wn-pages-plugin/blob/main/docs/documentation.md#custom-page-fields) and the dynamic syntax parser (https://wintercms.com/docs/v1.2/docs/services/parser#dynamic-syntax-parser) and the Blocks plugin (https://github.com/wintercms/wn-blocks-plugin) for more user friendly ways to edit page content.

In response to your question re: the PHP code section you provided:
> Static pages do not support having a PHP code section, that is what Components and the CMS pages are for.

Also your code would have to be

```twig
{% for (product in refined_products) %}
    {{ product.issue_date }}
{% endfor %}
```

**Reply by osamaahmed32** [reply]
How can I add components in pages can anyone guide me please

```
[viewBag]
title = "ABC Contracts"
url = "/abc-product-services/abc-contracts"
layout = "default"
is_hidden = 0
navigation_hidden = 0
==
<?php
use XXXX\Contracts\Models\Contracts;

    function onStart()
    {
        $this['contracts'] = Contracts::where('is_show', '=', '1' )->orderBy('id', 'asc')->get();
    }
?>
==

{% for contract in contracts %}
        { product_name: "{{ contract.title }}", product_symbol: "{{ contract.product_symbol }}", exchange: "{{ contract.exchange }}", product_group: "{{ contract.product_group }}", sub_group: "{{ contract.sub_group }}", settelement_type: "{{ contract.settlement_type }}", contract_summary: "{{ contract.summary_link }}" },
        {% endfor %}
```

this is not working 

just need help how to call or add dynamic content in static pages

Thanks alot in advance

**Reply by osamaahmed32** [reply]
```
[builderList consultationpapers]
modelClass = "XXXX\Consultationpapers\Models\ConsultationPapers"
scope = "-"
scopeValue = "{{ :consultation_paper }}"
displayColumn = "consultation_paper"
noRecordsMessage = "No records found"
detailsPage = "-"
detailsUrlParameter = "id"
recordsPerPage = "{{ 20 }}"
pageNumber = "{{ :page }}"
sortColumn = "consultation_paper"
sortDirection = "desc"
```

I have tried this but I dont know how can I use it properly

please guide me in this, how to do it properly

**Reply by LukeTowers** [reply]
Use the backend interface to add components to the page if you don't understand how to do it manually after reading the documentation: https://wintercms.com/docs/v1.2/docs/cms/components

**Comment by osamaahmed32** [comment]
I have done this using CMS content section

![image](https://github.com/user-attachments/assets/3b07add7-985a-41e3-b16d-e49ddcd40fa9)

and this in the code

```
[viewBag]
title = "Consultation Papers"
url = "/rules-regulations/consultation-papers"
layout = "default"
is_hidden = 0
navigation_hidden = 0

[builderList consultationpapers]
modelClass = "XXXX\Consultationpapers\Models\ConsultationPapers"
scope = "-"
scopeValue = "{{ :consultation_paper }}"
displayColumn = "consultation_paper"
noRecordsMessage = "No records found"
detailsPage = "-"
detailsUrlParameter = "id"
recordsPerPage = "{{ 20 }}"
pageNumber = "{{ :page }}"
sortColumn = "consultation_paper"
sortDirection = "desc"
==

```

then below in html using this code to display but nothing is working

```
{% component 'consultationpapers' %}
    {{ ConsultationPapers.consultationpapers.consultation_paper }}
    {{ :consultation_paper }}
```

**Reply by bennothommo** [reply]
You appear to be editing the static page content directly. Static pages don't support components - the key word being "static" in that it only supports static content. 

Only the "pages" and "layouts" in the CMS section support components.

Now if you were to use the `consultationpapers` component you have created above, you've attempted to call it in a couple of different ways, which have different requirements:

The `{% component 'consultationpapers' %}` call requires that you have specified [default markup content](https://wintercms.com/docs/v1.2/docs/cms/components#customizing-default-markup) for the component, otherwise, no content will show.

The `{{ ConsultationPapers.consultationpapers.consultation_paper }}` call - I don't even know if this will work, full stop. I believe that the component name is case-sensitive. However, if it were not case-sensitive, you would still need to have the component export a `consultationpapers` variable for that call to work. In that same [Default markup](https://wintercms.com/docs/v1.2/docs/plugin/components#default-markup) section in the docs, it details how to export a variable to the page through a component.

The `{{ :consultation_paper }}` call - I've never seen.

I'd recommend having a look at both of the pages below to get a good idea about how to use, and build, components:

- https://wintercms.com/docs/v1.2/docs/cms/components
- https://wintercms.com/docs/v1.2/docs/plugin/components

**Reply by osamaahmed32** [reply]
now i am doing like this 

```
{% if(this.page.title == 'Consultation Papers') %}
        {% partial 'consultation-papers' %}
        {% endif %}
```
        
 for different pages like this code i am adding in the layout and with js i am populating the dynamic data
 
 thank you all for all your help

**Comment by damsfx** [comment]
You can also use statics pages snippets to allow user to add dynamic content to static pages : 
> Snippets are elements that can be added by a Static Page, in the rich text editor. They allow to inject complex (and interactive) areas to pages. 

See: 
 - https://github.com/wintercms/wn-pages-plugin#snippets
 - https://github.com/wintercms/wn-pages-plugin/blob/main/docs/documentation.md#snippets-created-from-components
